src/lib/comms/sol-lwm2m-security.c

```diff
@@ -0,0 +1,470 @@
+    SOL_INT_CHECK(r, < 0, r);
+
+    if (!sol_network_link_addr_to_str(addr, &addr_str))
+        SOL_WRN("Could not convert the server address to string");
+    else
+        SOL_DBG("Looking for PSK ID for communication with server_id=%" PRId64
+            " and server_addr=%.*s", server_id,
+            SOL_STR_SLICE_PRINT(sol_buffer_get_slice(&addr_str)));
+
+    obj_ctx = find_object_ctx_by_id(lwm2m_client, SECURITY_OBJECT_ID);
+    if (!obj_ctx) {
+        SOL_WRN("LWM2M Security Object not provided!");
+        return -ENOENT;
+    }
+
+    if (!obj_ctx->instances.len) {
+        SOL_WRN("There are no Security Object instances");
+        return -ENOENT;
+    }
+
+    SOL_VECTOR_FOREACH_IDX (&obj_ctx->instances, instance, i) {
+        r = read_resources(lwm2m_client, obj_ctx, instance, res, 4,
+            SECURITY_SECURITY_MODE,
+            SECURITY_SERVER_ID,
+            SECURITY_PUBLIC_KEY_OR_IDENTITY,
+            SECURITY_IS_BOOTSTRAP);
+        SOL_INT_CHECK_GOTO(r, < 0, err_clear);
+
+        //If -ENOENT when reading SECURITY_PUBLIC_KEY_OR_IDENTITY or
+        // SECURITY_SECURITY_MODE != SOL_LWM2M_SECURITY_MODE_PRE_SHARED_KEY
+        if (res[2].data_len == 0 ||
+            res[0].data[0].content.integer != SOL_LWM2M_SECURITY_MODE_PRE_SHARED_KEY) {
+            clear_resource_array(res, sol_util_array_size(res));
+            continue;
+        }
+
+        if (res[1].data[0].content.integer == server_id ||
+            (res[3].data[0].content.b && (UINT16_MAX == server_id))) {
+            size_t psk_id_len;
+
+            if (res[2].data[0].content.blob->size > SOL_DTLS_PSK_ID_LEN) {
+                SOL_WRN("The PSK ID '%.*s' is %zu-bytes long;"
+                    " expecting a PSK ID at most %d-bytes long",
+                    SOL_STR_SLICE_PRINT(sol_str_slice_from_blob(res[2].data[0].content.blob)),
+                    res[2].data[0].content.blob->size, SOL_DTLS_PSK_ID_LEN);
+                r = -EINVAL;
+                goto err_clear;
+            }
+
+            psk_id_len = res[2].data[0].content.blob->size;
+
+            memcpy(id, res[2].data[0].content.blob->mem, psk_id_len);
+            clear_resource_array(res, sol_util_array_size(res));
+
+            return (ssize_t)psk_id_len;
+        }
+
+        clear_resource_array(res, sol_util_array_size(res));
+    }
+
+    return -ENOENT;
+
+err_clear:
+    clear_resource_array(res, sol_util_array_size(res));
+    return r;
+}
+
+static void
+sol_lwm2m_security_del_full(struct sol_lwm2m_security *security,
+    enum lwm2m_entity_type type)
+{
+    SOL_NULL_CHECK(security);
+
+    switch (type) {
+    case LWM2M_CLIENT:
+        ((struct sol_lwm2m_client *)(security->entity))->security = NULL;
+        break;
+    case LWM2M_SERVER:
+        ((struct sol_lwm2m_server *)(security->entity))->security = NULL;
+        break;
+    case LWM2M_BOOTSTRAP_SERVER:
+        ((struct sol_lwm2m_bootstrap_server *)(security->entity))->security = NULL;
+        break;
+    }
```

Comment:
So I can rely on the following check, mostly needed after Client- or Server-initiated Bootstrap: https://github.com/bsmelo/soletta/blob/lwm2m-dtls_psk-v1/src/lib/comms/sol-lwm2m-client.c#L2534.

Of course I would need to set only the `case LWM2M_CLIENT` to `NULL`, but then I thought it was better to keep them consistent.